module/Category/src/Controller/CategoryController.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace Category\Controller;
+
+use Zend\Mvc\Controller\AbstractActionController;
+use Zend\View\Model\ViewModel;
+use Doctrine\ORM\EntityManager;
+use Category\Repository\CategoryRepository;
+
+class CategoryController extends AbstractActionController
+{
+    private $entityManager;
+
+    public function __construct($serviceManager)
```

Comment:
Rather than injecting the `$serviceManager`, you should inject directly the services you need - in this case `$entityManager`, although really you should create your own service in this module and inject that (with `$entityManager` injected into that service, instead of into this controller).

Injecting the `$serviceManager` is considered to be an anti-pattern.